merkle/src/merkle.rs

```diff
@@ -674,6 +676,10 @@ impl<T: Element, A: Algorithm<T>, K: Store<T>> MerkleTree<T, A, K> {
         self.data.read_at(i)
     }
 
+    pub fn read_into(&self, pos: usize, buf: &mut [u8]) {
+        self.data.read_at(pos).copy_to_slice(buf);
```

Comment:
I guess this is also generating an unnecessary copy, I'll move it at the `Store` level again.

Reply:
The `Element` abstraction forces me (I think, not sure) to make a copy, the `Store` should figure out ranges and copy directly bypassing it, this optimization won't be pretty.